core/string.cpp

```diff
@@ -57,38 +59,55 @@ std::map<DType, std::string> &dtypeNameRegistry() {
   return registry;
 }
 
-const std::string to_iso_date(const scipp::core::time_point &item,
-                              const units::Unit &unit) {
-  int64_t ts = item.time_since_epoch();
+namespace {
+template <class Ratio> constexpr int64_t num_digits() {
+  static_assert(Ratio::num == 1 || Ratio::num % 10 == 0);
+  static_assert(Ratio::den == 1 || Ratio::den % 10 == 0);
+  static_assert(Ratio::den > Ratio::num);
+  int64_t result = 0;
+  for (std::size_t i = Ratio::num; i < Ratio::den; i *= 10) {
+    ++result;
+  }
+  return result;
+}
+
+// For synchronizing access to gmtime because its return value is shared.
+std::mutex gmtime_mutex;
+
+void put_time(std::ostream &os, const std::time_t time_point) {
+  std::lock_guard guard_{gmtime_mutex};
+  const std::tm *tm = std::gmtime(&time_point);
```

Comment:
Can this pointer be stored in a static variable, to avoid having to lock the mutex on each invokation of this?

Reply:
How would this help? All threads potentially store a pointer to the same piece of memory. Every time anyone calls `std::gmtime` (or another function from that family!) it overwrites that memory. And it makes no difference how the pointer is stored.

Reply:
Is this the recommend usage then? It feels weird that we need a mutex for formatting datetimes?

Reply:
I don't know about recommended usage. It is an old c functions from before there was widespread multi threading support.